sim/midas/src/main/cc/bridges/synthesized_prints.h

```diff
@@ -3,123 +3,114 @@
-        void flush();
-        void finish() { flush(); };
-    private:
-        PRINTBRIDGEMODULE_struct * mmio_addrs;
-        const unsigned int print_count;
-        const unsigned int  token_bytes;
-        const unsigned int  idle_cycles_mask;
-        const unsigned int* print_offsets;
-        const char* const*  format_strings;
-        const unsigned int* argument_counts;
-        const unsigned int* argument_widths;
-        const unsigned int stream_idx;
-        const unsigned int stream_depth;
-        ClockInfo clock_info;
-        const int printno;
-
-        // DMA batching parameters
-        const size_t beat_bytes  = DMA_DATA_BITS / 8;
-        // The number of DMA beats to pull off the FPGA on each invocation of tick()
-        // This will be set based on the ratio of token_size : desired_batch_beats
-        size_t batch_beats;
-        // This will be modified to be a multiple of the token size
-        const size_t desired_batch_beats = stream_depth / 2;
-
-        // Used to define the boundaries in the batch buffer at which we'll
-        // initalize GMP types
-        using gmp_align_t = uint64_t;
-        const size_t gmp_align_bits = sizeof(gmp_align_t) * 8;
-
-
-        // +arg driven members
-        std::ofstream printfile;   // Used only if the +print-file arg is provided
-        std::string default_filename = "synthesized-prints.out";
-
-        std::ostream* printstream; // Is set to std::cerr otherwise
-        uint64_t start_cycle, end_cycle; // Bounds between which prints will be emitted
-        uint64_t current_cycle = 0;
-        bool human_readable = true;
-        bool print_cycle_prefix = true;
-
-        std::vector<std::vector<size_t>> widths;
-        std::vector<size_t> sizes;
-        std::vector<print_vars_t*> masks;
-
-        std::vector<size_t> aligned_offsets; // Aligned to gmp_align_t
-        std::vector<size_t> bit_offset;
-
-        bool current_print_enabled(gmp_align_t* buf, size_t offset);
-        size_t process_tokens(size_t beats, size_t minimum_batch_beats);
-        void show_prints(char * buf);
-        void print_format(const char* fmt, print_vars_t* vars, print_vars_t* masks);
-        // Returns the number of beats available, once two successive reads return the same value
-        int beats_avaliable_stable();
+class synthesized_prints_t : public bridge_driver_t {
+
+public:
+  synthesized_prints_t(
+      simif_t *sim, std::vector<std::string> &args,
+      PRINTBRIDGEMODULE_struct *mmio_addrs, unsigned int print_count,
+      unsigned int token_bytes, unsigned int idle_cycles_mask,
+      const unsigned int *print_offsets, const char *const *format_strings,
+      const unsigned int *argument_counts, const unsigned int *argument_widths,
+      unsigned int stream_idx, unsigned int stream_depth,
+      const char *const clock_domain_name, const unsigned int clock_multiplier,
+      const unsigned int clock_divisor, int printno);
```

Comment:
I'll be the first to admit that the signature of this constructor is comically long but `clang-format` has made this almost unreadable by smashing collapsing it.

Reply:
I think we can adjust this in clang-format.

Reply:
We could add
```
BinPackArguments: false
BinPackParameters: false
```
to ensure func args are either all on one line, or all on their own line.

Reply:
I tried these options and the output doesn't look great. We can either roll with what we have now or I can try to find some other options, although that would introduce further delays since I don't quite have time for that now.

Reply:
@nandor can you be more specific about what didn't look great?  When I scanned through the output with those options enabled, it looked like the output was only changing the arguments to be either all on one line or all on their own line.

Reply:
```
fprintf(stderr,
          "Some format %x\n",
          number);
```
is one example

Reply:
We could bin-pack arguments to function calls, placing parameters on new lines.

Reply:
Thanks for providing the detail.